File: src/types/actions.ts
export type Actions =
  | 'action.bridge.command.advance'
  | 'action.bridge.command.cannon'
  | 'action.bridge.command.prime'
  | 'action.bridge.command.mobilize'
  | 'action.bridge.command.reinforce'
  | 'action.bridge.command.liberate'
  | 'action.bridge.command.analyze'
  | 'action.bridge.comms.propa'
  | 'action.bridge.comms.music'
  | 'action.bridge.comms.onair'
  | 'action.bridge.comms.rally'
  | 'action.bridge.comms.hail'
  | 'action.bridge.comms.interceptsignal'
  | 'action.common-area.socialize'
  | 'action.engine-room.calibrate'
  | 'action.engine-room.shield'
  | 'action.expocrafts-bay.ready'
  | 'action.expocrafts-bay.unready'
  | 'action.expocrafts-bay.repair'
  | 'action.fightercrafts-bay.calibrate'
  | 'action.fightercrafts-bay.dogfight'
  | 'action.fightercrafts-bay.patrolrun'
  | 'action.fightercrafts-bay.repair'
  | 'action.flak-turret.shoot'
  | 'action.laser-turret.shoot'
  | 'action.medlab.compound'
  | 'action.medlab.firstaid'
  | 'action.medlab.surgery'
  | 'action.garden.forage'
  | 'action.garden.grow'
  | 'action.garden.harvest'
  | 'action.generic.shower'
  | 'action.refectory.provision'
  | 'action.refectory.consume'
  | 'action.rnd.review'
  | 'action.rnd.research'
  | 'action.rnd.build'
  | 'action.rnd.repair'
  | 'action.private-quarters.sleep'
  | 'action.private-quarters.seggs'
  | 'action.weapon-system.calibrate'

export interface ActionDamage {
  min: number
  max: number
}

File: src/nakama/gamelogic/types/actions.go
package types

type Action string

const (
	// Bridge Command actions
	ActionBridgeCommandAdvance   Action = "action.bridge.command.advance"
	ActionBridgeCommandCannon    Action = "action.bridge.command.cannon"
	ActionBridgeCommandPrime     Action = "action.bridge.command.prime"
	ActionBridgeCommandMobilize  Action = "action.bridge.command.mobilize"
	ActionBridgeCommandReinforce Action = "action.bridge.command.reinforce"
	ActionBridgeCommandLiberate  Action = "action.bridge.command.liberate"
	ActionBridgeCommandAnalyze   Action = "action.bridge.command.analyze"

	// Bridge Comms actions
	ActionBridgeCommsPropa           Action = "action.bridge.comms.propa"
	ActionBridgeCommsMusic           Action = "action.bridge.comms.music"
	ActionBridgeCommsOnair           Action = "action.bridge.comms.onair"
	ActionBridgeCommsRally           Action = "action.bridge.comms.rally"
	ActionBridgeCommsHail            Action = "action.bridge.comms.hail"
	ActionBridgeCommsInterceptsignal Action = "action.bridge.comms.interceptsignal"

	// Common Area actions
	ActionCommonAreaSocialize Action = "action.common-area.socialize"

	// Engine Room actions
	ActionEngineRoomCalibrate Action = "action.engine-room.calibrate"
	ActionEngineRoomShield    Action = "action.engine-room.shield"

	// ExpoCrafts Bay actions
	ActionExpoCraftsBayReady   Action = "action.expocrafts-bay.ready"
	ActionExpoCraftsBayUnready Action = "action.expocrafts-bay.unready"
	ActionExpoCraftsBayRepair  Action = "action.expocrafts-bay.repair"
	ActionExpoCraftsBayLaunch  Action = "action.expocrafts-bay.launch"

	// FighterCrafts Bay actions
	ActionFighterCraftsBayCalibrate Action = "action.fightercrafts-bay.calibrate"
	ActionFighterCraftsBayDogfight  Action = "action.fightercrafts-bay.dogfight"
	ActionFighterCraftsBayPatrolrun Action = "action.fightercrafts-bay.patrolrun"
	ActionFighterCraftsBayRepair    Action = "action.fightercrafts-bay.repair"
	ActionFighterCraftsBayCAS       Action = "action.fightercrafts-bay.cas"

	// Turret actions
	ActionFlakTurretShoot  Action = "action.flak-turret.shoot"
	ActionLaserTurretShoot Action = "action.laser-turret.shoot"

	// Medlab actions
	ActionMedlabCompound Action = "action.medlab.compound"
	ActionMedlabFirstaid Action = "action.medlab.firstaid"
	ActionMedlabSurgery  Action = "action.medlab.surgery"

	// Garden actions
	ActionGardenForage  Action = "action.garden.forage"
	ActionGardenGrow    Action = "action.garden.grow"
	ActionGardenHarvest Action = "action.garden.harvest"

	// Generic actions
	ActionGenericShower  Action = "action.generic.shower"
	ActionGenericEquip   Action = "action.generic.equip"
	ActionGenericUnequip Action = "action.generic.unequip"

	// Refectory actions
	ActionRefectoryProvision Action = "action.refectory.provision"
	ActionRefectoryConsume   Action = "action.refectory.consume"

	// RND actions
	ActionRndReview   Action = "action.rnd.review"
	ActionRndResearch Action = "action.rnd.research"
	ActionRndBuild    Action = "action.rnd.build"
	ActionRndRepair   Action = "action.rnd.repair"

	// Private Quarters actions
	ActionPrivateQuartersSleep Action = "action.private-quarters.sleep"
	ActionPrivateQuartersSeggs Action = "action.private-quarters.seggs"

	// Weapon System actions
	ActionWeaponSystemCalibrate Action = "action.weapon-system.calibrate"

	// Storage actions
	ActionStorageStash    Action = "action.storage.stash"
	ActionStorageRetrieve Action = "action.storage.retrieve"
)

